public/services/saved_objects/saved_object_savers/ppl/save_as_new_vis.ts

```diff
@@ -76,9 +83,14 @@ export class SaveAsNewVisualization extends SavedQuerySaver {
   }
 
   addToPanel({ selectedPanels, saveTitle, notifications, visId }) {
+    const { dispatch } = this.dispatchers;
```

Comment:
Please consider context here, as well as SRP-SingleResponsibilityPrinciple.

Wer are inside "saved_objects/saved_object_savers".... This, despite it's name, is the Observability index re-invention of saved objects.
We don't want to add any code here.  This whole module will (eventually) be deprecated/removed.

Also, wrt. SRP,  although "saving things" sounds like a single responsibliity, we can see from the code that saving to SavedObject-Core (via Redux/Dispatch) is _very_ different from using SavedObjectSavers.
These are different responsibilities.


Recommendation : 
Lift the filter/split of saved-object (UUID) and Observability (non-UUID) out of this method.  Pass the non-UUID into this method, and specifically call dispatch from the caller (which is a component, which is insdie React context, which has correct context for `useDispatch()` ).    This will also remove the prop-drilling of "dispatchers".

Reply:
> we can see from the code that saving to SavedObject-Core (via Redux/Dispatch) is very different from using SavedObjectSavers

Why does this happen? I didn't look into impl details but essentially the difference between saving SO vs. observability object is the client. If the client can be abstracted, then the same operation should work for both cases

Reply:
@joshuali925 @pjfitzgibbons can we merge this in as is? I understand it may not be right but theres a few other p0 items today, once we make the final go-no go call I can refactor to be correct